### PR TITLE
Fixes "Mention the difference between stable docs and latest docs."

### PR DESCRIPTION
**Pull request by bt3gl**
(try to) Solve #1996 

I am not sure whether this is the best way, so please advise if you have a different idea (see comment in the original issue).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -26,6 +26,8 @@ Documentation is available at `cirq.readthedocs.io <https://cirq.readthedocs.io>
 - `Documentation <https://cirq.readthedocs.io>`_
 - `Tutorial <https://cirq.readthedocs.io/en/stable/tutorial.html>`_
 
+Note: The (*default*) documentation for ``stable`` might be behind ``latest`` (see `here <https://cirq.readthedocs.io/en/latest/tutorial.html>`_), which can cause confusion if you are working on branches such as ``cirq-dev`` or ``cirq-release``.
```
Comment (vtomole) [inline README.rst]: Mention what `stable` is (the docs that we just linked above).

Comment (vtomole) [reply README.rst]: People who read this for the first time won't know what `cirq-dev` and `cirq-release` are.

Comment (bt3gl) [reply README.rst]: What should I say about stable? I looked at the docs and didn't see any explicit explanation on how you folks are releasing stable.

Something like this?

Note: ``stable`` is ???. Therefore, the documentation for ``stable`` 
might be behind ``latest`` (see `here <https://cirq.readthedocs.io/en/latest/tutorial.html>`_).

Comment (vtomole) [reply README.rst]: What do you think about moving this sentence up? 

Documentation is available at cirq.readthedocs.io.  The (*default*) documentation for [stable](https://cirq.readthedocs.io/en/stable/) might be behind [latest](https://cirq.readthedocs.io/en/latest/tutorial.html).

Comment (bt3gl) [reply README.rst]: Done!

**Comment by bt3gl** [comment]
Well, more info is always better. Doc is updated with this last suggestion. We are getting close :).

**Comment by bt3gl** [comment]
@cduck Good catches, and done.

File: README.rst [hunk 2]
```diff
@@ -20,12 +20,19 @@ circuits and running them against quantum computers and simulators.
 Installation and Documentation
 ------------------------------
 
-Documentation is available at `cirq.readthedocs.io <https://cirq.readthedocs.io>`_:
+Cirq documentation is available at `cirq.readthedocs.io <https://cirq.readthedocs.io>`_.
+
+Documentation for the latest **unstable** version of cirq (tracks the repository's master branch; what you get if you ``pip install cirq-unstable``), is available at `cirq.readthedocs.io/latest <https://cirq.readthedocs.io/en/latest/>`_.
```
Comment (mpharrigan) [inline README.rst]: extra credit: you could configure readthedocs to make the url `/unstable/` instead of `/latest/` for consistent terminology

**Pull request merged**